packages/components/src/ConditionalFilter/TextFilter.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, Fragment } from 'react';
+import { TextInput } from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import './conditional-filter.scss';
+
+interface TextProps {
+  /** Optional text filter value. */
+  value?: string;
+  /** Optional text filter placeholder. */
+  placeholder?: string;
+  /** Optional onChange event called on input change. */
+  onChange?: (e: object, value: string) => void;
+  /** Optional onSubmit event called on Enter press. */
+  onSubmit?: (e: object, value: string) => void;
+  /** Optional input disabled state indicator. */
+  isDisabled?: boolean;
+  /** Optional aria-label value. */
+  'aria-label'?: string;
+  /** Optional id value. */
+  id?: string;
+  /** Optional icon component. */
+  icon?: React.FunctionComponent;
+  /** Optional className. */
+  className?: string;
+}
+
+/**
+ * Component that works as a text filter for ConditionalFilter component.
+ *
+ * This component accepts onChange event called on input change
+ * and onSubmit event called on Enter press.
+ */
+const Text: React.FunctionComponent<TextProps> = ({ value, onChange, onSubmit, id, icon, className, isDisabled, ...props }) => {
+  const [stateValue, setStateValue] = useState('');
+  const Icon = icon || SearchIcon;
+  const changeCallback = (e: object, value: string) => (onChange ? onChange(e, value) : setStateValue(value));
+
+  return (
+    <Fragment>
+      <TextInput
+        {...props}
```

Comment:
If you are sending the rest of the props the `TextProps` must extend the `TextInputProps` provided by PF.

Reply:
It would be maybe a better and cleaner solution to not sent the rest of the props - current `onChange` and `onSubmit` functions do not comply with the PF ones

Reply:
It would, but It may introduce regressions. We can revisit the definition later. We have to be carefoul.